### DISCUSSION TITLE
Working Battery/Trackpad/Volume keys

### DISCUSSION THREAD

**Discussion by backdash756** [badstorm/surface-pro-7-opencore#50] [Q&A]
Thank you for the great work @badstorm 

I really don't know how I did it, but after a few unsuccessful touchscreen attempts with VoodooI2c.kext and VoodooI2cGoodix.kext, the following works perfectly:

- Battery status
- Trackpad recognized in the system settings and fully functional
- Volume up/down buttons work now.

Maybe you can draw conclusions from my EFI folder:
https://github.com/backdash756/surface-pro-7-opencore/releases/tag/1776

<img width="1368" alt="Bildschirmfoto 2022-01-20 um 09 13 31" src="https://user-images.githubusercontent.com/98048832/150313883-63a42593-903c-4a94-9d50-a60bc73e4017.png">
<img width="1368" alt="Bildschirmfoto 2022-01-20 um 09 06 04" src="https://user-images.githubusercontent.com/98048832/150313913-454c681e-2c25-43b8-9c54-07de27bd0841.png">
.

**Comment by badstorm** [comment]
Thanks @backdash756 for this update. Please send a PR with your changes so we can disscusse there.
